### ISSUE TITLE
Members of method parameter are not resolved when type is trait's 'associated type'  

### ISSUE THREAD

**Issue by Exidex** [intellij-rust/intellij-rust#4940]
## Environment

* **IntelliJ Rust plugin version:** 0.2.115.2157-193
* **Rust toolchain version:** 1.41.0
* **CLion:** 2019.3.3
Build #CL-193.6015.37, built on January 20, 2020
Runtime version: 11.0.5+10-b520.30 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 5.3.18-1-MANJARO

## Problem description
Members of method parameter are not resolved when type is trait's 'associated type'  

## Steps to reproduce
The issue could probably be reproduced with simpler code but that's how I was able to reproduce it. Following code compiles just fine. Only function call is unresolved.
![Screenshot from 2020-02-08 00-11-05](https://user-images.githubusercontent.com/16986685/74072587-8aac5e00-4a07-11ea-9ac6-b6b7a7a3b95e.png)

<details><summary>Code as plain text</summary>
<p>

```rust
trait NotResolvedTrait {
    fn errored_fn(self);
}
trait TraitWithAssociatedType {
    type AssociatedType: NotResolvedTrait;
}

struct TheStruct;

trait TheTrait<B: TraitWithAssociatedType> {
    fn fn_with(&mut self, event: B::AssociatedType);
}

impl<B: TraitWithAssociatedType> TheTrait<B> for TheStruct {
    fn fn_with(&mut self, event: B::AssociatedType) {
        event.errored_fn();
    }
}
```
</p>
</details>

**Comment by Undin** [comment]
@vlad20012 could you take a look?

**Comment by vlad20012** [comment]
Associated type bounds are not supported for now, but I have a branch... that I again forgot to push

**Comment by vlad20012** [comment]
Fixed by #4974
